File: Atividades-SistemasDistribuidos/Unidade-II/3-ProblemasDeProgramacao/Ex9/src/main/java/ex9/rmi/Servidor09RMI.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ex9.rmi;

import home09.BaralhoInterfaceRMI;
import home09.NaipeCarta;
import home09.ValorCarta;
import java.rmi.registry.LocateRegistry;
import java.rmi.registry.Registry;
import java.rmi.server.UnicastRemoteObject;

/**
 *
 * @author mathe
 */
public class Servidor09RMI implements BaralhoInterfaceRMI {
    private int valor;
    private int naipe;

    public String nomeCarta(int valor, int naipe) {
        setNaipe(naipe);
        setValor(valor);

        return obterValorCarta() + " de " + obterNaipeCarta();
    }

    public String obterNaipeCarta() {
        String strNaipeCarta = "";
        switch (getNaipe()) {
            case 1:
                strNaipeCarta += NaipeCarta.OUROS;
                break;
            case 2:
                strNaipeCarta += NaipeCarta.PAUS;
                break;
            case 3:
                strNaipeCarta += NaipeCarta.COPAS;
                break;
            case 4:
                strNaipeCarta += NaipeCarta.ESPADAS;
                break;
            default:
                break;

        }
        return strNaipeCarta;
    }

    public String obterValorCarta() {
        String strValorCarta = "";
        switch (getValor()) {
            case 1:
                strValorCarta += ValorCarta.AS;
                break;
            case 2:
                strValorCarta += ValorCarta.DOIS;
                break;
            case 3:
                strValorCarta += ValorCarta.TRES;
                break;
            case 4:
                strValorCarta += ValorCarta.QUATRO;
                break;
            case 5:
                strValorCarta += ValorCarta.CINCO;
                break;
            case 6:
                strValorCarta += ValorCarta.SEIS;
                break;
            case 7:
                strValorCarta += ValorCarta.SETE;
                break;
            case 8:
                strValorCarta += ValorCarta.OITO;
                break;
            case 9:
                strValorCarta += ValorCarta.NOVE;
                break;
            case 10:
                strValorCarta += ValorCarta.DEZ;
                break;
            case 11:
                strValorCarta += ValorCarta.VALETE;
                break;
            case 12:
                strValorCarta += ValorCarta.DAMA;
                break;
            case 13:
                strValorCarta += ValorCarta.REI;
                break;
            default:
                break;
        }

        return strValorCarta;
    }

    public int getValor() {
        return valor;
    }

    public void setValor(int valor) {
        this.valor = valor;
    }

    public int getNaipe() {
        return naipe;
    }

    public void setNaipe(int naipe) {
        this.naipe = naipe;
    }
    
    public static void main(String[] args) {
        try {
            Servidor09RMI obj = new Servidor09RMI();
            BaralhoInterfaceRMI stub = (BaralhoInterfaceRMI) UnicastRemoteObject.exportObject(obj, 0);
            LocateRegistry.createRegistry(1099);
            Registry registry = LocateRegistry.getRegistry();
            registry.bind("InterfaceRMI", stub);

            System.out.println("ServidorRMI vinculado no registro");
        } catch (Exception e) {
            System.err.println("Server exception: " + e.toString());
            e.printStackTrace();
        }
    }
}

File: Atividades-SistemasDistribuidos/Unidade-II/2-Lista01PythonIterativo/Ex9/rmi/Servidor09RMI.py
import Pyro4

from Ex9.home09.Baralho import Baralho
from Ex9.home09.NaipeCarta import NaipeCarta
from Ex9.home09.ValorCarta import ValorCarta


@Pyro4.expose
class Servidor09RMI(object):

    def nomeCarta(self, valor, naipe):
        strNaipeCarta = ""

        if (naipe == 1):
            strNaipeCarta = strNaipeCarta, NaipeCarta.OUROS.value
        elif (naipe == 2):
            strNaipeCarta = strNaipeCarta, NaipeCarta.PAUS.value
        elif (naipe == 3):
            strNaipeCarta = strNaipeCarta, NaipeCarta.COPAS.value
        elif (naipe == 4):
            strNaipeCarta = strNaipeCarta, NaipeCarta.ESPADAS.value

        strValorCarta = ""

        if (valor == 1):
            strValorCarta = strValorCarta, ValorCarta.AS.value
        elif (valor == 2):
            strValorCarta = strValorCarta, ValorCarta.DOIS.value
        elif (valor == 3):
            strValorCarta = strValorCarta, ValorCarta.TRES.value
        elif (valor == 4):
            strValorCarta = strValorCarta, ValorCarta.QUATRO.value
        elif (valor == 5):
            strValorCarta = strValorCarta, ValorCarta.CINCO.value
        elif (valor == 6):
            strValorCarta = strValorCarta, ValorCarta.SEIS.value
        elif (valor == 7):
            strValorCarta = strValorCarta, ValorCarta.SETE.value
        elif (valor == 8):
            strValorCarta = strValorCarta, ValorCarta.OITO.value
        elif (valor == 9):
            strValorCarta = strValorCarta, ValorCarta.NOVE.value
        elif (valor == 10):
            strValorCarta = strValorCarta, ValorCarta.DEZ.value
        elif (valor == 11):
            strValorCarta = strValorCarta, ValorCarta.VALETE.value
        elif (valor == 12):
            strValorCarta = strValorCarta, ValorCarta.DAMA.value
        elif (valor == 13):
            strValorCarta = strValorCarta, ValorCarta.REI.value

        return strValorCarta, " de ", strNaipeCarta

daemon = Pyro4.Daemon.serveSimple({Servidor09RMI: 'Servidor09RMI', }, host="localhost", port=8000, ns=False, verbose=True)
ns = Pyro4.locateNS()
server = Servidor09RMI()
uri = daemon.register(server)
ns.register('serverRMI', uri)
print('Server ...')
daemon.requestLoop()